src/osd/SnapMapper.cc

```diff
@@ -99,13 +100,11 @@ int OSDriver::get_next(
 
 string SnapMapper::get_prefix(int64_t pool, snapid_t snap)
 {
-  char buf[100];
-  int len = snprintf(
-    buf, sizeof(buf),
-    "%lld_%.*X_",
-    (long long)pool,
-    (int)(sizeof(snap)*2), static_cast<unsigned>(snap));
-  return MAPPING_PREFIX + string(buf, len);
+  assert(sizeof(snap) == 8); // 2*8 is the '16' in the '%.16X' format below
```

Comment:
Hmm, the sake of consistency suggests `ceph_assert()`. I guess this shouldn't inflate the check's penalty as (IIRC; please check me) we already define `assert` to be an alias to `ceph_assert`.

Reply:
^H^H^H

How about `static_assert` instead?

Reply:
Yes. It should have been static_assert. Will fix.